### ISSUE TITLE
Concurrency when using TestServer

### ISSUE THREAD

**Issue by ericywl** [JosephLenton/axum-test#78]
## Problem

I was trying to do some concurrency testing on my app and ran into this issue. The following is a simplified snippet:

```rs
#[tokio::test]
async fn test_something() {
    let app = axum::Router::new().route(
        &"/todo",
        axum::routing::get(|| async {
            tokio::time::sleep(std::time::Duration::from_secs(1)).await;
            "hello!"
        }),
    );
    let server = axum_test::TestServer::new(app).unwrap();

    let future1 = async { server.get("/todo").await };
    let future2 = async { server.get("/todo").await };
    let (r1, r2) = tokio::join!(future1, future2);

    assert_eq!(r1.text(), r2.text());
}
```

It seems this code does not halt, not sure if its due to some internal workings of the `TestServer` or if I did something wrong.

## Environment

Cargo.toml:
```toml
[dependencies]
axum = { version = "0.7.5", features = ["macros"] }
axum-test = "14.10.0"
tokio = { version = "1.38.0", features = ["full"] }
```

**Comment by JosephLenton** [comment]
Hey, I have just spent some time looking at this. I am pretty certain this is a bug. I think this is a deadlock bug.

For more context; internally there is a `TransportLayer`, and this is wrapped in a lock. I think it gets locked for the first request, that then waits, the second request then goes to grab the lock, is also blocked (first still holds the lock), and so everything just halts.

I will look into fixing this today.

**Comment by JosephLenton** [comment]
Hey @ericywl , you should now find this fixed in the latest version [15.0.0](https://crates.io/crates/axum-test/15.0.0). This version also brings a requirement for using Rust 1.75 or higher.

Please try that version, and feel free to let me know if there are any issues.

**Comment by ericywl** [comment]
@JosephLenton Thanks for the quick response, I have confirmed that it's working as expected. Closing the issue now.
